guava/src/com/google/common/io/LineReader.java

```diff
@@ -84,4 +91,31 @@ public String readLine() throws IOException {
     }
     return lines.poll();
   }
+
+  /**
+   * Returns a {@code Stream}, the elements of which are lines read from
+   * this {@code LineReader}.
+   *
+   * @return a {@code Stream<String>} providing the lines of text
+   *     described by this {@code LineReader}
```

Comment:
I believe the JavaDoc should:

- state that the caller is in charge of closing the Stream
- mention the fact that we throw UncheckedIOException
- what we define as a line (maybe just mention that a line has the same semantics as in LineReader#lines